sanity.config.ts

```diff
@@ -40,6 +46,14 @@ export default defineConfig({
   // edit schemas in './src/schemas'
   schema: {
     types: schemaTypes,
+    templates: (templates) =>
+      templates.filter(({ schemaType }) => !singletonTypes.has(schemaType)),
+  },
+  document: {
+    actions: (input, context) =>
+      singletonTypes.has(context.schemaType)
+        ? input.filter(({ action }) => action && singletonActions.has(action))
+        : input,
```

Comment:
Allow only predefined actions for the singleton documents (publish, discard, restore)